pkg/scheduler/scheduling_queue.go

```diff
@@ -482,9 +482,26 @@ func (sq *SchedulingQueue) getHeadRoom() *resources.Resource {
 	if sq.parent != nil {
 		parentHeadRoom = sq.parent.getHeadRoom()
 	}
+	return sq.internalHeadRoom(parentHeadRoom)
+}
+
+// this function returns the max headRoom of a queue
+// this doesn't get the partition resources into the consideration
+func (sq *SchedulingQueue) getMaxHeadRoom() *resources.Resource {
```

Comment:
Maybe I missed something here, I think we don't need getMaxHeadRoom (and it should be Headroom instead of HeadRoom).

What we really need is to be able to pass the parent's Headroom (instead of getting from parent.Headroom, which is be used by scheduling).

Also, this patch doesn't take hiearchical resource quota into account. We can do this better by doing the following logic. 

```
// Return total assigned resource of this queue
queue.getQueueOutstandingRequests(parentHeadroom, outstandingRequests) *Resource{
	myHeadroom = queue.getHeadRoom(parentHeadroom)

	totalOutstandingResource := 0

	if queue.isLeafQueue() {
       for _, app := range sq.sortApplications() {
       	  app.getOutstandingRequests(myHeadRoom, total)
       }
	} else {

	   for child : queue.children() {
	       assigned := child.getQueueOutstandingRequests(myHeadroom, outstandingRequests)

	       myHeadroom -= assigned

	       totalOutstandingResource += assigned 

	       if myHeadroom <= 0 {
	           return totalOutstandingResource
	       }
	   }
	}
}

// And for root, it passes nil for parentHeadroom.
```

Reply:
I do not see how this would solve the hierarchy issue.
It might even cause more issues as the headroom for a queue might be different for different locations in the queue hierarchy. The normal headroom includes the max from the root (i.e cluster size) so we cannot use that and need to use the max as defined below to start with.

Reply:
based on the discussion, we don't have to solve this for now. correct?

Reply:
You cannot solve this without the changes as proposed in #173.
Without that making any changes here does not help. So no, it just will not work correctly in a queue hierarchy.